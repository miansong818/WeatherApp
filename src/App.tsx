import { useEffect, useState } from "react";
import "./App.css";
import Today from "./components/Today";
import { Coordinates, CurrentWeather, Daily, Hourly } from "./Types";
import NextDays from "./components/NextDays";
import { GetDates } from "./utils";
import Header from "./components/Header";
import {
  MockDailyData,
  MockCurrentWeather,
  MockHourlyData,
} from "./mockData/MockWeather";
import { ErrorBoundary } from "react-error-boundary";
import Error from "./components/Error";

const initCurrentWeather: CurrentWeather = MockCurrentWeather;

const hours: Hourly[] = MockHourlyData;
const nextDays: Daily[] = MockDailyData;

function App() {
  const [coordinates, setCoordinates] = useState({ lat: 0, lon: 0 });
  const [currentWeather, setCurrentWeather] = useState(initCurrentWeather);
  const [nextHoursData, setNextHoursData] = useState(hours);
  const [nextDaysData, setNextDaysData] = useState(nextDays);
  const dayCount = 16;
  const dateList = GetDates(dayCount);

  // Fetch the weather location API, update the current weather state
  const fetchWeatherForCurrentLocation = async () => {
    const { lat, lon } = coordinates;
    await fetch(
      `${process.env.VITE_API_URL}/weather/?lat=${lat}&lon=${lon}&units=metric&APPID=${process.env.VITE_API_KEY}`
    )
      .then((res) => res.json())
      .then((result) => {
        setCurrentWeather(result);
      });
  };

  //get the current location coordinates on mount
  const getCurrentLocation = async () => {
    navigator.geolocation.getCurrentPosition((position) => {
      setCoordinates({
        lat: position.coords.latitude,
        lon: position.coords.longitude,
      });
    });

    //
    await fetchWeatherForCurrentLocation();
  };

  //
  const fetchData = async () => {
    const { lat, lon } = coordinates;
    await fetch(
      `${process.env.VITE_API_URL}/onecall?lat=${lat}&lon=${lon}&units=metric&APPID=${process.env.VITE_API_KEY}`
    )
      .then((res) => res.json())
      .then((result) => {
        setNextHoursData(result.hourly);
        setNextDaysData(result.daily);
      });
  };

  //Update current weather when location coordinates updates
  useEffect(() => {
    fetchWeatherForCurrentLocation();
    fetchData();
  }, [coordinates]);

  // set weather data on mount with the current location
  useEffect(() => {
    getCurrentLocation;
    fetchData();
  }, []);

  const handleInputChange = (result: Coordinates) => {
    setCoordinates(result);
  };

  return (
    <>
      <Header searchFn={handleInputChange} />
      <ErrorBoundary fallback={<Error />}>
        <Today
          currentData={currentWeather}
          nextHoursData={nextHoursData}
        ></Today>

        <NextDays nextDaysData={nextDaysData} dateList={dateList}></NextDays>
      </ErrorBoundary>
    </>
  );
}

export default App;
