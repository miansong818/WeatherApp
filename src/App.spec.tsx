import { render } from "@testing-library/react";
import { describe, it } from "node:test";
import App from "./App";
import "../__mocks__/matchMedia.js";

describe("App rendering", () => {
  // Added the matchMedia to the setup-test.ts file, is equivalent to mock beforeAll()

  test("demo", () => {
    expect(true).toBe(true);
  });

  it("Renders App", () => {
    render(<App />);
  });

  // const { asFragment } = render(<App />);

  // expect(asFragment()).toMatchSnapshot();
});
