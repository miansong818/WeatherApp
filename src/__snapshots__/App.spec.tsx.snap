// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div
    class="relative isolate overflow-hidden bg-gray-900 py-12 sm:py-22"
  >
    <div
      aria-hidden="true"
      class="hidden sm:absolute sm:-top-10 sm:right-1/2 sm:-z-10 sm:mr-10 sm:block sm:transform-gpu sm:blur-3xl"
    >
      <div
        class="aspect-[1097/845] w-[68.5625rem] bg-gradient-to-tr from-[#ff4694] to-[#776fff] opacity-20"
        style="clip-path: polygon(74.1% 44.1%, 100% 61.6%, 97.5% 26.9%, 85.5% 0.1%, 80.7% 2%, 72.5% 32.5%, 60.2% 62.4%, 52.4% 68.1%, 47.5% 58.3%, 45.2% 34.5%, 27.5% 76.7%, 0.1% 64.9%, 17.9% 100%, 27.6% 76.8%, 76.1% 97.7%, 74.1% 44.1%);"
      />
    </div>
    <div
      aria-hidden="true"
      class="absolute -top-52 left-1/2 -z-10 -translate-x-1/2 transform-gpu blur-3xl sm:top-[-28rem] sm:ml-16 sm:translate-x-0 sm:transform-gpu"
    >
      <div
        class="aspect-[1097/845] w-[68.5625rem] bg-gradient-to-tr from-[#ff4694] to-[#776fff] opacity-20"
        style="clip-path: polygon(74.1% 44.1%, 100% 61.6%, 97.5% 26.9%, 85.5% 0.1%, 80.7% 2%, 72.5% 32.5%, 60.2% 62.4%, 52.4% 68.1%, 47.5% 58.3%, 45.2% 34.5%, 27.5% 76.7%, 0.1% 64.9%, 17.9% 100%, 27.6% 76.8%, 76.1% 97.7%, 74.1% 44.1%);"
      />
    </div>
    <div
      class="mx-auto max-w-7xl px-6 lg:px-8"
    >
      <div
        class="mx-auto max-w-2xl lg:mx-0"
      >
        <h2
          class="text-4xl font-bold tracking-tight text-white sm:text-6xl"
        >
          Today's weather
        </h2>
      </div>
    </div>
    <div
      class="lg:flex lg:items-center lg:justify-between"
    >
      <div
        class="min-w-0 flex-1"
      >
        <div
          class="mt-1 flex flex-col sm:mt-0 sm:(flex-row flex-wrap space-x-6)"
        />
      </div>
      <div
        class="m-5 flex lg:(m-4 mt-0)"
      >
        <div>
          <input
            class="min-w-0 w-80 flex-auto rounded-md border-0 bg-white/5 px-3.5 text-white shadow-sm ring-1 ring-inset ring-white/10 focus:ring-2 focus:ring-inset focus:ring-indigo-500 sm:text-sm sm:leading-9"
            id="searchBar"
            name="search"
            placeholder="Search..."
            type="text"
            value=""
          />
          <ul>
            <li
              class="flex gap-x-4 bg-slate-200 text-black cursor-pointer hover:bg-slate-500"
            >
              <span />
              <span />
              <span />
            </li>
          </ul>
        </div>
        <span
          class="ml-3  sm:block"
        >
          <button
            class="inline-flex items-center rounded-md bg-white px-3 py-2 text-sm font-semibold text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 hover:bg-gray-50"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="-ml-0.5 mr-1.5 h-5 w-5 text-gray-400"
              data-slot="icon"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M9 3.5a5.5 5.5 0 1 0 0 11 5.5 5.5 0 0 0 0-11ZM2 9a7 7 0 1 1 12.452 4.391l3.328 3.329a.75.75 0 1 1-1.06 1.06l-3.329-3.328A7 7 0 0 1 2 9Z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </span>
      </div>
    </div>
  </div>
  <div
    class="bg-slate-50 py-24 sm:py-32 overflow-hidden"
  >
    <div
      class="mx-auto grid max-w-7xl gap-x-8 gap-y-20 px-6 lg:px-8 xl:grid-cols-2"
    >
      <div
        class="max-w-2xl text-center"
      >
        <h2
          class="mb-5 text-3xl font-bold tracking-tight text-gray-900 sm:text-4xl"
        >
          My location
        </h2>
        <h3
          class="mt-6 text-4xl leading-normal  text-gray-600"
        >
          Ivyland
        </h3>
        <p
          class="h mt-6 text-8xl leading-normal text-gray-600"
        >
          19째
        </p>
        <p
          class="mt-6 text-xl leading-8 text-gray-600"
        >
          L: 17.71 H: 19.66
        </p>
        <p
          class="mt-6 text-lg leading-8 text-gray-600"
        >
          Mist
        </p>
      </div>
      <div
        class="max-w-lg lg:max-w-2xl m-auto"
      >
        <div
          class="m-10 justtify-center"
        >
          <div
            class="slick-slider slick-initialized"
            dir="ltr"
          >
            <button
              class="slick-arrow slick-prev"
              data-role="none"
              style="display: block;"
              type="button"
            >
               Previous
            </button>
            <div
              class="slick-list"
            >
              <div
                class="slick-track"
                style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
              >
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="-4"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="flex text-center justify-center gap-x-2 rounded-md border-solid border-2 text-gray-900"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="justify-center flex"
                      >
                        <img
                          alt=""
                          class="h-16 w-16 rounded-full"
                          src="https://openweathermap.org/img/wn/01n@2x.png"
                        />
                      </div>
                      <div>
                        <h3
                          class="text-base font-semibold leading-7 tracking-tight text-gray-400"
                        >
                          Time: 03:00 AM
                        </h3>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          humidity: 32
                        </p>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          Temp: 27째
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="-3"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="flex text-center justify-center gap-x-2 rounded-md border-solid border-2 text-gray-900"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="justify-center flex"
                      >
                        <img
                          alt=""
                          class="h-16 w-16 rounded-full"
                          src="https://openweathermap.org/img/wn/01n@2x.png"
                        />
                      </div>
                      <div>
                        <h3
                          class="text-base font-semibold leading-7 tracking-tight text-gray-400"
                        >
                          Time: 04:00 AM
                        </h3>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          humidity: 33
                        </p>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          Temp: 27째
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="-2"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="flex text-center justify-center gap-x-2 rounded-md border-solid border-2 text-gray-900"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="justify-center flex"
                      >
                        <img
                          alt=""
                          class="h-16 w-16 rounded-full"
                          src="https://openweathermap.org/img/wn/01n@2x.png"
                        />
                      </div>
                      <div>
                        <h3
                          class="text-base font-semibold leading-7 tracking-tight text-gray-400"
                        >
                          Time: 05:00 AM
                        </h3>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          humidity: 34
                        </p>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          Temp: 27째
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="-1"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="flex text-center justify-center gap-x-2 rounded-md border-solid border-2 text-gray-900"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="justify-center flex"
                      >
                        <img
                          alt=""
                          class="h-16 w-16 rounded-full"
                          src="https://openweathermap.org/img/wn/01n@2x.png"
                        />
                      </div>
                      <div>
                        <h3
                          class="text-base font-semibold leading-7 tracking-tight text-gray-400"
                        >
                          Time: 06:00 AM
                        </h3>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          humidity: 35
                        </p>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          Temp: 27째
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="false"
                  class="slick-slide slick-active slick-current"
                  data-index="0"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="flex text-center justify-center gap-x-2 rounded-md border-solid border-2 text-gray-900"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="justify-center flex"
                      >
                        <img
                          alt=""
                          class="h-16 w-16 rounded-full"
                          src="https://openweathermap.org/img/wn/01n@2x.png"
                        />
                      </div>
                      <div>
                        <h3
                          class="text-base font-semibold leading-7 tracking-tight text-gray-400"
                        >
                          Time: Now
                        </h3>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          humidity: 31
                        </p>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          Temp: 27째
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="false"
                  class="slick-slide slick-active"
                  data-index="1"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="flex text-center justify-center gap-x-2 rounded-md border-solid border-2 text-gray-900"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="justify-center flex"
                      >
                        <img
                          alt=""
                          class="h-16 w-16 rounded-full"
                          src="https://openweathermap.org/img/wn/01n@2x.png"
                        />
                      </div>
                      <div>
                        <h3
                          class="text-base font-semibold leading-7 tracking-tight text-gray-400"
                        >
                          Time: 03:00 AM
                        </h3>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          humidity: 32
                        </p>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          Temp: 27째
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="false"
                  class="slick-slide slick-active"
                  data-index="2"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="flex text-center justify-center gap-x-2 rounded-md border-solid border-2 text-gray-900"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="justify-center flex"
                      >
                        <img
                          alt=""
                          class="h-16 w-16 rounded-full"
                          src="https://openweathermap.org/img/wn/01n@2x.png"
                        />
                      </div>
                      <div>
                        <h3
                          class="text-base font-semibold leading-7 tracking-tight text-gray-400"
                        >
                          Time: 04:00 AM
                        </h3>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          humidity: 33
                        </p>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          Temp: 27째
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="false"
                  class="slick-slide slick-active"
                  data-index="3"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="flex text-center justify-center gap-x-2 rounded-md border-solid border-2 text-gray-900"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="justify-center flex"
                      >
                        <img
                          alt=""
                          class="h-16 w-16 rounded-full"
                          src="https://openweathermap.org/img/wn/01n@2x.png"
                        />
                      </div>
                      <div>
                        <h3
                          class="text-base font-semibold leading-7 tracking-tight text-gray-400"
                        >
                          Time: 05:00 AM
                        </h3>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          humidity: 34
                        </p>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          Temp: 27째
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide"
                  data-index="4"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="flex text-center justify-center gap-x-2 rounded-md border-solid border-2 text-gray-900"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="justify-center flex"
                      >
                        <img
                          alt=""
                          class="h-16 w-16 rounded-full"
                          src="https://openweathermap.org/img/wn/01n@2x.png"
                        />
                      </div>
                      <div>
                        <h3
                          class="text-base font-semibold leading-7 tracking-tight text-gray-400"
                        >
                          Time: 06:00 AM
                        </h3>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          humidity: 35
                        </p>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          Temp: 27째
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="5"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="flex text-center justify-center gap-x-2 rounded-md border-solid border-2 text-gray-900"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="justify-center flex"
                      >
                        <img
                          alt=""
                          class="h-16 w-16 rounded-full"
                          src="https://openweathermap.org/img/wn/01n@2x.png"
                        />
                      </div>
                      <div>
                        <h3
                          class="text-base font-semibold leading-7 tracking-tight text-gray-400"
                        >
                          Time: Now
                        </h3>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          humidity: 31
                        </p>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          Temp: 27째
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="6"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="flex text-center justify-center gap-x-2 rounded-md border-solid border-2 text-gray-900"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="justify-center flex"
                      >
                        <img
                          alt=""
                          class="h-16 w-16 rounded-full"
                          src="https://openweathermap.org/img/wn/01n@2x.png"
                        />
                      </div>
                      <div>
                        <h3
                          class="text-base font-semibold leading-7 tracking-tight text-gray-400"
                        >
                          Time: 03:00 AM
                        </h3>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          humidity: 32
                        </p>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          Temp: 27째
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="7"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="flex text-center justify-center gap-x-2 rounded-md border-solid border-2 text-gray-900"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="justify-center flex"
                      >
                        <img
                          alt=""
                          class="h-16 w-16 rounded-full"
                          src="https://openweathermap.org/img/wn/01n@2x.png"
                        />
                      </div>
                      <div>
                        <h3
                          class="text-base font-semibold leading-7 tracking-tight text-gray-400"
                        >
                          Time: 04:00 AM
                        </h3>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          humidity: 33
                        </p>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          Temp: 27째
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="8"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="flex text-center justify-center gap-x-2 rounded-md border-solid border-2 text-gray-900"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="justify-center flex"
                      >
                        <img
                          alt=""
                          class="h-16 w-16 rounded-full"
                          src="https://openweathermap.org/img/wn/01n@2x.png"
                        />
                      </div>
                      <div>
                        <h3
                          class="text-base font-semibold leading-7 tracking-tight text-gray-400"
                        >
                          Time: 05:00 AM
                        </h3>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          humidity: 34
                        </p>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          Temp: 27째
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="9"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="flex text-center justify-center gap-x-2 rounded-md border-solid border-2 text-gray-900"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="justify-center flex"
                      >
                        <img
                          alt=""
                          class="h-16 w-16 rounded-full"
                          src="https://openweathermap.org/img/wn/01n@2x.png"
                        />
                      </div>
                      <div>
                        <h3
                          class="text-base font-semibold leading-7 tracking-tight text-gray-400"
                        >
                          Time: 06:00 AM
                        </h3>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          humidity: 35
                        </p>
                        <p
                          class="text-sm font-semibold leading-6 text-indigo-600"
                        >
                          Temp: 27째
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <button
              class="slick-arrow slick-next"
              data-role="none"
              style="display: block;"
              type="button"
            >
               Next
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <section
    class="relative isolate overflow-hidden bg-white px-6 py-24 sm:py-32 lg:px-8"
  >
    <div
      class="absolute inset-0 -z-10 bg-[radial-gradient(45rem_50rem_at_top,theme(colors.indigo.100),white)] opacity-20"
    />
    <div
      class="absolute inset-y-0 right-1/2 -z-10 mr-16 w-[200%] origin-bottom-left skew-x-[-30deg] bg-white shadow-xl shadow-indigo-600/10 ring-1 ring-indigo-50 sm:mr-28 lg:mr-0 xl:mr-16 xl:origin-center"
    />
    <div
      class="mx-auto max-w-2xl lg:max-w-4xl"
    >
      <ul
        class="max-w-3xl divide-y divide-gray-100"
        role="list"
      >
        <li
          class="justify-between grid grid-cols-4 gap-4 py-5 text-center place-items-center"
        >
          <img
            alt="Clear"
            class="h-12 w-12 flex-none rounded-full bg-gray-50"
            src="https://openweathermap.org/img/wn/01d@2x.png"
          />
          <p
            class="text-lg font-semibold leading-6 text-gray-900"
          >
            Friday
          </p>
          <p
            class="text-sm leading-6 text-gray-900"
          >
            29째 / 27째
          </p>
          <p
            class="text-sm leading-6 text-gray-900"
          >
            clear sky
          </p>
        </li>
        <li
          class="justify-between grid grid-cols-4 gap-4 py-5 text-center place-items-center"
        >
          <img
            alt="Rain"
            class="h-12 w-12 flex-none rounded-full bg-gray-50"
            src="https://openweathermap.org/img/wn/10d@2x.png"
          />
          <p
            class="text-lg font-semibold leading-6 text-gray-900"
          >
            Saturday
          </p>
          <p
            class="text-sm leading-6 text-gray-900"
          >
            28째 / 27째
          </p>
          <p
            class="text-sm leading-6 text-gray-900"
          >
            light rain
          </p>
        </li>
        <li
          class="justify-between grid grid-cols-4 gap-4 py-5 text-center place-items-center"
        >
          <img
            alt="Rain"
            class="h-12 w-12 flex-none rounded-full bg-gray-50"
            src="https://openweathermap.org/img/wn/10d@2x.png"
          />
          <p
            class="text-lg font-semibold leading-6 text-gray-900"
          >
            Sunday
          </p>
          <p
            class="text-sm leading-6 text-gray-900"
          >
            28째 / 27째
          </p>
          <p
            class="text-sm leading-6 text-gray-900"
          >
            light rain
          </p>
        </li>
        <li
          class="justify-between grid grid-cols-4 gap-4 py-5 text-center place-items-center"
        >
          <img
            alt="Rain"
            class="h-12 w-12 flex-none rounded-full bg-gray-50"
            src="https://openweathermap.org/img/wn/10d@2x.png"
          />
          <p
            class="text-lg font-semibold leading-6 text-gray-900"
          >
            Monday
          </p>
          <p
            class="text-sm leading-6 text-gray-900"
          >
            28째 / 27째
          </p>
          <p
            class="text-sm leading-6 text-gray-900"
          >
            light rain
          </p>
        </li>
        <li
          class="justify-between grid grid-cols-4 gap-4 py-5 text-center place-items-center"
        >
          <img
            alt="Clear"
            class="h-12 w-12 flex-none rounded-full bg-gray-50"
            src="https://openweathermap.org/img/wn/01d@2x.png"
          />
          <p
            class="text-lg font-semibold leading-6 text-gray-900"
          >
            Tuesday
          </p>
          <p
            class="text-sm leading-6 text-gray-900"
          >
            28째 / 27째
          </p>
          <p
            class="text-sm leading-6 text-gray-900"
          >
            clear sky
          </p>
        </li>
      </ul>
    </div>
  </section>
</DocumentFragment>
`;
